knotx-core/knotx-knot-action/src/test/java/com/cognifide/knotx/knot/action/ActionKnotVerticleTest.java

```diff
@@ -58,27 +69,110 @@
 
   @Test
   @KnotxConfiguration("knotx-knot-action-test.json")
-  public void callWithNoActionFragments_expectResponseOkNoFragmentChanges(TestContext context) throws Exception {
+  public void callGetWithNoActionFragments_expectResponseOkNoFragmentChanges(TestContext context) throws Exception {
     String firstFragment = "<html><head></head><body>";
     String lastFragment = "</body></html>";
-    String expectedFormFragment = FileReader.readText("form_step1_out_fragment.txt");
+    String expectedTempatingFragment = FileReader.readText("fragment_templating_out.txt");
 
     KnotContext knotContext = KnotContext.empty(Arrays.asList(
         Fragment.raw(firstFragment),
-        Fragment.snippet("form-identifier", FileReader.readText("form_step1_in_fragment.txt")),
+        Fragment.snippet("templating", FileReader.readText("fragment_templating_in.txt")),
         Fragment.raw(lastFragment)));
 
     callActionKnotWithAssertions(context, knotContext,
         clientResponse -> {
           context.assertEquals(HttpResponseStatus.OK, knotContext.clientResponse().statusCode());
           List<Fragment> fragments = knotContext.fragments().get();
           context.assertEquals(firstFragment, fragments.get(0).getContent());
-          context.assertEquals(expectedFormFragment, fragments.get(1).getContent());
+          context.assertEquals(expectedTempatingFragment, fragments.get(1).getContent());
           context.assertEquals(lastFragment, fragments.get(2).getContent());
 
         },
         error -> context.fail(error.getMessage()));
+  }
+
+  @Test
+  @KnotxConfiguration("knotx-knot-action-test.json")
+  public void callGetWithTwoActionFragments_expectResponseOkTwoFragmentChanges(TestContext context) throws Exception {
+    String firstFragment = "<html><head></head><body>";
+    String lastFragment = "</body></html>";
+    String expectedRedirectFormFragment = clean(FileReader.readText("fragment_form_redirect_out.txt"));
+    String expectedSelfFormFragment = clean(FileReader.readText("fragment_form_self_out.txt"));
+
+    KnotContext knotContext = KnotContext.empty(Arrays.asList(
```

Comment:
Knotcontext.empty() is missleading. I'd rather create it using fluent api of context.
new KnotContext().setFragments(...).setRaw(..).etc